internal/bodyprocessors/json.go

```diff
@@ -18,26 +18,52 @@ type jsonBodyProcessor struct{}
 var _ plugintypes.BodyProcessor = &jsonBodyProcessor{}
 
 func (js *jsonBodyProcessor) ProcessRequest(reader io.Reader, v plugintypes.TransactionVariables, _ plugintypes.BodyProcessorOptions) error {
+	// Read the entire body to store it and process it
+	s := strings.Builder{}
+	tr := io.TeeReader(reader, &s)
+
+	// Process as normal
 	col := v.ArgsPost()
-	data, err := readJSON(reader)
+	data, err := readJSON(tr)
 	if err != nil {
 		return err
 	}
 	for key, value := range data {
 		col.SetIndex(key, 0, value)
 	}
+
+	// Store the raw JSON in the TX variable for validateSchema
+	// This is needed because RequestBody is a Single interface without a Set method
+	if txVar := v.TX(); txVar != nil && v.RequestBody() != nil {
```

Comment:
I am not sure if any of this checks will work because here we are talking about interfaces. See for example https://go.dev/play/p/8AjGRzDJ9Vb

Reply:
Updated to remove unnecessary nil checks. The first check will work (and breaks tests if it is removed as that's tested,) but good catch as the others were not necessary.